android/src/main/kotlin/com/instrumentisto/medea_flutter_webrtc/controller/RtpSenderController.kt

```diff
@@ -82,12 +86,18 @@ class RtpSenderController(messenger: BinaryMessenger, private val sender: RtpSen
         result.success(null)
       }
       "dispose" -> {
-        chan.setMethodCallHandler(null)
+        dispose()
         result.success(null)
       }
     }
   }
 
+  override fun dispose() {
+    sender.setDisposed()
+    chan.setMethodCallHandler(null)
+    ControllerRegistry.unregister(this)
```

Comment:
In some places controller unregisters in `ControllerRegistry` after doing disposing logic, but in some cases we're doing it before. There can be issues if some step of disposing logic will throw exception. What we want to do in this case? Should it be marked as disposed or not?

I think it will be better to unregister it at the start of `dispose()`, so it will not be stored in `ControllerRegistry` even if it failed to dispose properly. And this dispose exception will be catched in `ControllerRegistry.disposeAll` and logged as error anyway.